### PR TITLE
Configurable source folders

### PR DESCRIPTION
**Pull request by atomfrede**
This enables to use a custom source folder layout (#32). Furthermore following updates are done:

* Update galting to latest `2.3.0` along with scala to `2.12.3`
* Scala version is now configurable, in case you want to use an older gatling version
* I tried to update the build tooling to gradle `4.1` but I ran into serious problems, as the unit tests couldn't resolve the plugin anymore
* 4 tests (running gatling) are failing locally, but they are failing on master too. In particular apache commons can't be resolved during gatling execution
* If you just add the plugin the usual auto detection of source layout is not changed, so convention over configuration still applies

The new configuration looks like this:

```groovy
gatling {
    toolVersion = '2.2.2'
    scalaVersion = '2.11.3'
    sourceRoot = 'src/test/gatling`
    simulationsDir = 'user-files/simulations'
    dataDir = 'user-files/data' 
    bodiesDir = 'user-files/bodies' 
    confDir = 'conf' 
}
```

EDIT: Sorry seems like master works, so will check why commons is not imported to gatling compile classpath

### CODE REVIEW

**Comment by eshepelyuk** [comment]
May we still  have `Scala` 2.11 ? 
2.12 is quite new and some ppls are using this plugin together with their sources that is still 2.11

**Comment by atomfrede** [comment]
At least gatling 2.13 was updated to scala 2.12, that why I put the default 2.12: https://groups.google.com/forum/#!topic/gatling/-dyUahNGD6g

**Comment by eshepelyuk** [comment]
Is latest `Gatling` still supports both 2.11 and 2.12 ?

**Comment by atomfrede** [comment]
The tests are failing because there is something changed with logback. There is this check:

```
and: "logs doesn't contain INFO"
!result.output.split().any { it.contains("INFO") }
```

but somehow multiple lines like `21:02:12,051 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]`

will check whats the problem is. When removing that check from the test everything is fine.

**Comment by atomfrede** [comment]
Found the logback problem. The `immediateFlush` property must be at a different place for logback 1.2 which gatling 2.3 is using. I made a switch for users using older gatling versions not to break their setup. See this commit for details: https://github.com/gatling/gatling/commit/6a35dbaa8607deda4a20abf1e8a553d9cd9cef01

File: src/main/groovy/com/github/lkishalmi/gradle/gatling/LogbackConfigTaskAction.groovy [hunk 1]
```diff
@@ -23,10 +39,21 @@ public class LogbackConfigTaskAction implements Action<Task> {
     void execute(Task gatlingRunTask) {
         def gatlingExt = gatlingRunTask.project.extensions.getByType(GatlingPluginExtension)
         if (!gatlingRunTask.project.file("${gatlingExt.confDir()}/logback.xml").exists()) {
-            new File(gatlingRunTask.project.buildDir, "resources/gatling/logback.xml").with {
-                parentFile.mkdirs()
-                text = template(gatlingExt.logLevel)
+
+            // gatling 2.3.x updated logback and the config file has changed therefore we must differentiate between >= 2.3 and <2.3
+            // see https://github.com/gatling/gatling/commit/6a35dbaa8607deda4a20abf1e8a553d9cd9cef01 for details
+            if (gatlingExt.toolVersion.startsWith("2.3")) {
```
Comment (atomfrede) [inline src/main/groovy/com/github/lkishalmi/gradle/gatling/LogbackConfigTaskAction.groovy]: Maybe it is better to do a check for a number as this becomes quite hard to maintain when gatling updates to 2.4 for example.

Comment (eshepelyuk) [reply src/main/groovy/com/github/lkishalmi/gradle/gatling/LogbackConfigTaskAction.groovy]: Hmm, I believe you may ignore that completely.

**Comment by eshepelyuk** [comment]
Please revert all the changes that are not actual **changes**, i.e. code formatting, tabs, spaces etc

**Comment by eshepelyuk** [comment]
Instead of introducint this check everywhere `isAutoDetect` - plz just introduce proper **default** values. 
so then `isAutoDetect` can be removed.

**Comment by atomfrede** [comment]
Are you sure you can define proper defaults, as there is this automatic to set folders. So when setting `sourceRoot = src/gatling` we may do something like this `"${sourceRoot}${isGatlingLayout ? "" : "/resources"}/${dataDir}" but it looks strange for the simulations folder: `"${sourceRoot}${isGatlingLayout ? "simulations" : "${simulationsDir"}"` (and the default for `simulationsDir` is `scala` in that case, but need to test if it works like the current solution.

**Pull request merged**